internal/cmd/root.go

```diff
@@ -36,7 +37,26 @@ func init() {
 		settings.PersistChanges()
 	}
 	rootCmd.CompletionOptions.HiddenDefaultCmd = true
-
+	rootCmd.PersistentPostRun = func(cmd *cobra.Command, args []string) {
```

Comment:
Aren't you you overriding the rootCmd.PersistentPostRun set above in line 35/36 that calls `settings.PersistentChanges()`